openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -83,15 +84,22 @@ def main(date_start: str = "all", date_end: str = None):
 def _add_data_to_buffer(**kwargs):
     endpoint = _create_endpoint_for_IDs(**kwargs)
     IDs = _get_image_IDs(endpoint)
+    if IDs is None:
+        return
 
     for id_ in IDs:
-        if id_ is not None:
-            details = _get_meta_data(id_)
-            if details is not None:
-                image_store.add_item(**details)
+        item_data = _get_object_json(id_)
+        if item_data:
+            _process_item(item_data)
 
 
-def _get_total_images():
+def _process_item(item_data: Dict) -> None:
```

Comment:
I extracted this method so that I can use it in tests with the saved json responses.